numpy/core/numeric.py

```diff
@@ -1457,6 +1461,93 @@ def rollaxis(a, axis, start=0):
     return a.transpose(axes)
 
 
+def _validate_axis(axis, ndim, argname):
+    try:
+        axis = [operator.index(axis)]
+    except TypeError:
+        axis = list(axis)
+    original_axis = axis
+    axis = [a + ndim if a < 0 else a for a in axis]
+    if not builtins.all(0 <= a < ndim for a in axis):
+        raise ValueError('%s=%s has elements outside valid bounds' %
+                         (argname, original_axis))
+    if len(set(axis)) != len(axis):
+        raise ValueError('repeated axis in %s=%s' % (argname, original_axis))
+    return axis
+
+
+def moveaxis(a, source, destination):
+    """
+    Move axes of an array to new positions.
```

Comment:
On the other hand, it somewhat depends on the types `transpose` implementation. I suppose it is not guaranteed to be a view for non-ndarray types.